### PR TITLE
avm2: Implement `StaticText.text`

### PR DESCRIPTION
**Pull request by kjarosh**
* Fixes https://github.com/ruffle-rs/ruffle/issues/12120

### CODE REVIEW

File: core/src/font.rs [hunk 1]
```diff
@@ -806,6 +810,9 @@ pub struct Glyph {
 
     shape: GlyphShape,
     advance: Twips,
+
+    // The character this glyph represents.
+    character: char,
```
Comment (evilpie) [inline core/src/font.rs:815]: Maybe `Option<char>` instead?

File: core/src/avm2/globals/flash/text/static_text.rs [hunk 2]
```diff
@@ -16,9 +17,17 @@ pub fn static_text_allocator<'gc>(
...
 ) -> Result<Value<'gc>, Error<'gc>> {
-    avm2_stub_getter!(activation, "flash.text.StaticText", "text");
-    Ok(activation.strings().empty().into())
+    if let Some(this) = this
+        .as_object()
+        .and_then(|this| this.as_display_object())
+        .and_then(|this| this.as_text())
+    {
+        let text = this.text(activation.context);
+        return Ok(AvmString::new(activation.context.gc_context, text).into());
```
Comment (evilpie) [inline core/src/avm2/globals/flash/text/static_text.rs:29]: nit: `activation.gc()`

**Review by evilpie — APPROVED** [review]

File: core/src/avm2/globals/flash/text/static_text.rs (hunk 2 above)
Comment (kjarosh) [reply core/src/avm2/globals/flash/text/static_text.rs]: Right!

File: core/src/font.rs (hunk 1 above)
Comment (kjarosh) [reply core/src/font.rs:817]: If you're referring to the replacement character, I don't believe `Option<char>` is appropriate here, as it seems that Flash just doesn't care whether `u16` is a surrogate code point or not (which is the requirement behind Rust's `char`). However, the replacement character is used everywhere, so I just did the same and added a TODO comment (just now). I believe we need a wider refactor of Flash characters here, but that's outside of this PR.

**Pull request merged**